tests/tests_e3/python/main_test.py

```diff
@@ -149,3 +150,46 @@ def test_star_requirements():
         pypi.add_wheel(wheel3.path)
         pypi.add_requirement("src2==1.0.0")
         assert len(pypi.closure()) == 2
+
+
+@pytest.mark.parametrize(
+    "arguments,expected",
+    [
+        ((None, None), "setuptools_scm-7.1.0-py3-none-any.whl"),
+        ((["setuptools-scm"], None), "setuptools_scm-8.0.0-py3-none-any.whl"),
+        ((["setuptools_scm"], None), "setuptools_scm-8.0.0-py3-none-any.whl"),
+        ((None, "setuptools_scm==8"), None),
+    ],
+)
+def test_yanked(pypi_server, arguments, expected):
+    allowed_yanked, invalid_wheel = arguments
+    expected_wheel = expected
+
+    rm("cache", recursive=True)
```

Comment:
Are you sure we have to remove them ? I mean that it should be a new tests started, thus its own temp folder.

Reply:
I'm not sure, I just thought that when in doubt it would be more robust and avoid mistakes. The cache is used to avoid operations that I want to test.

Reply:
I confirm that each test (aka for each set of parameters) is run in its own tmp directory. You may want to test :)